vm/chef/cookbooks/singlefs/files/singlefs-install

```diff
@@ -0,0 +1,296 @@
+      if (( "${LOCAL_SSDS}" > "1" )); then
+        echo "Add local ssds as caching devices"
+        zpool add "${STORAGE_POOL}" cache -f ${DISK_PREFIX}-local-ssd-?
+      fi
+    fi
+    ;;
+  ext4)
+    # Use largest possible ext4 journal for performance.
+    # TODO(leffler): Add logic based on disk size.
+    echo "y" | mkfs.ext4 -J size=1000 -L "${STORAGE_POOL}" "${DATA_DEVICE}"
+    echo "LABEL=${STORAGE_POOL} /${STORAGE_POOL} ext4 defaults 0 0" >> /etc/fstab
+    ;;
+  xfs)
+    mkfs.xfs -L "${STORAGE_POOL}" "${DATA_DEVICE}"
+    echo "LABEL=${STORAGE_POOL} /${STORAGE_POOL} xfs defaults 0 0" >> /etc/fstab
+    ;;
+esac
+
+if [[ "${FILE_SYSTEM}" != "zfs" ]]; then
+  mkdir "/${STORAGE_POOL}"
+  mount "/${STORAGE_POOL}"
+fi
+
+# Relax permissions.
+chmod 777 "/${STORAGE_POOL}"
+
+# Share the filesystem to local GCE network over NFS.
+# TODO(leffler): Add bash logic to find CIDRs of subnetworks.
+if [[ "${FILE_SYSTEM}" = "zfs" ]]; then
+  sharenfs=""
+  for network in "${networks[@]}"; do
+    sharenfs="rw=${network},fsid=100,${sharenfs}"
+  done
+  sharenfs="${sharenfs}insecure"
+  zfs set "sharenfs=${sharenfs}" "${STORAGE_POOL}"
+  # Fix issue related to ZFS exports without /etc/exports.
+  sed -i "s/if \[ -f \/etc\/exports \]/if \[ -f \/etc\/exports \] #/" \
+    /etc/init.d/nfs-kernel-server
+sed -i "s/ZFS_SHARE='no'/ZFS_SHARE='yes'/" /etc/default/zfs
+else
+  for network in "${networks[@]}"; do
+    echo "/${STORAGE_POOL} ${network}(rw,no_subtree_check,fsid=100)" >> /etc/exports
+  done
+fi
+
+if is_centos; then
+  NFS_CONFIG="/etc/sysconfig/nfs"
+  NFS_SERVICES=(nfs-server.service)
+  SAMBA_SERVICES=(smb.service nmb.service)
+fi
+if is_debian; then
+  NFS_CONFIG="/etc/default/nfs-kernel-server"
+  NFS_SERVICES=(nfs-kernel-server)
+  SAMBA_SERVICES=(smbd nmbd)
+  # Fix issue related to ZFS exports without /etc/exports.
+  sed -i "s/if \[ -f \/etc\/exports \]/if \[ -f \/etc\/exports \] #/" \
+    /etc/init.d/nfs-kernel-server
+fi
+
+# Use 16 * (# cores) NFS threads. This provides good performance for both
+# streaming and random IO without overwhelming the CPU with thread scheduling.
+readonly NFS_THREADS=$(($(nproc) * 16 ))
+sed -i "s/#*RPCNFSDCOUNT=.*/RPCNFSDCOUNT=${NFS_THREADS}/" "${NFS_CONFIG}"
+
+# Use a fixed port number for mountd, so that the port does not change
+# during failover.
+sed -i "s/RPCMOUNTDOPTS=\"/RPCMOUNTDOPTS=\"--port 2050 /" "${NFS_CONFIG}"
+
+# Start and enable NFS. Don't fail installation if NFS doesn't return
+# succesfully on start. The validation script checks if NFS started correctly.
+if [[ "${ENABLE_NFS}" = "enable:True" ]]; then
+  for service in "${NFS_SERVICES[@]}"; do
+    systemctl enable "${service}" || fail "Failed to enable service: ${service}"
+    systemctl start "${service}" || true
+  done
+fi
+
+# Configure SMB.
+cp /etc/samba/smb.conf /etc/samba/smb.conf.orig
+cat >> /etc/samba/smb.conf << EOF
+# Public share added by click to deploy for testing.
```

Comment:
Is removal of this share an action required from the end user? Is yes, is it covered in next steps presented after installation?